docs/tutorials/09_Protein_Folding.ipynb

```diff
@@ -0,0 +1,708 @@
wyjrg8q5Y8cOoVAoxIYNG8SFCxfERx99JKytrcWRI0fke+jj+5gJkJ599NFHolmzZkKpVIonnnhCHDt2zNQhPRQANX5iY2Plc27duiVeffVV0ahRI+Hg4CAGDx4sMjMzTRe0ntydAJlTOb/77jvRrl07YWtrKwIDA8WGDRu0jqvVajFnzhzh7e0tbG1tRc+ePUVycrKJoq27goICMWXKFNGsWTNhZ2cnWrZsKd5++22tP44NsawHDhyo8b/LyMhIIUTtynTjxg0xcuRI4eTkJFxcXMS4ceNEYWGhCUpzf/cra2pq6j2/ow4cOCDfwxzKWpOaEqCGUNbalHPjxo0iICBA2NnZifbt24uvv/5a6x76+D6WhKgyJSoRERGRBWAfICIiIrI4TICIiIjI4jABIiIiIovDBIiIiIgsDhMgIiIisjhMgIiIiMjiMAEiIiIii8MEiIiIiCwOEyAiojravHkz3NzcTB0GEdUBEyAiuqf+/fujd+/eNR47cuQIJElCUlKSvG/Lli14/PHH4eDgAGdnZ/To0QNxcXFa1x08eBCSJNX4ycrKqvFZly9fhiRJSExM1LrH3atBG1Lz5s2xatUqrX3Dhw/Hn3/+abQYiEh/mAAR0T1FRUUhPj4eV69erXYsNjYWnTp1QkhICABg+vTpePnllzF8+HAkJSXh+PHj6N69OwYOHIg1a9ZUuz45ORmZmZlaHy8vL4OXqSohBCoqKup8vb29vdFjJiI9eehVzYjIbJWXlwtvb2+xaNEirf2FhYXCyclJrFu3TgghREJCggAgPvzww2r3mDZtmlAoFCItLU0IcWchxL///rvWcWgWvTx9+nSNC2BqFlFUqVTi3XffFc2bNxd2dnYiJCREbN++Xb6P5tk//PCD+Mc//iEUCoU4cOCASElJEQMGDBBeXl7C0dFRdOrUScTHx8vX9ejRo9ozhRAiNjZWuLq6asX68ccfi5YtWwqFQiEeffRRsXXrVq3jAMSnn34qBg0aJOzt7UVAQID45ptv5OO5ubnihRdeEI0bNxZ2dnYiICBAbNq0qdbviohqhwkQEd3XjBkzRKtWrYRarZb3bdq0Sdjb24u8vDwhhBCTJ08WTk5OWqusa1y7dk0AkFetftgEqKKiQvzvf/8TAERycrLIzMyU43jnnXdEYGCg+PHHH8XFixdFbGyssLW1FQcPHtR6dkhIiNizZ49ISUkRN27cEImJiWL9+vXizJkz4s8//xSzZ88WdnZ24sqVK0KIyhW2mzRpIhYuXCgyMzPlVafvToB27NghFAqFWLt2rUhOThbvv/++sLa2Fvv375fPASCaNGkitm3bJi5cuCC/uxs3bgghhIiJiREdOnQQv/76q0hNTRXx8fHi22+/rfW7IqLaYQJERPd1/vx5AUAcOHBA3vfkk0+KF198Ud7u3bu3aN++/T3v4eLiIqKjo4UQd5IQR0dHrU9QUNA9r6+aAFW9R9UkqqSkRDg4OIijR49qXRsVFSVGjhypdd3XX3/9wHK3bdtWfPTRR/K2v7+/nMRp3J0Ade3aVUyYMEHrnGHDhom+ffvK2wDE7Nmz5e2ioiIBQOzatUsIIUT//v3FuHHjHhgfET0cG2M1tRFRwxQYGIiuXbti06ZNeOqpp5CSkoIjR45g4cKFWucJIe57H6VSqbV95MgRODs7y9sKheKh4kxJScHNmzfx7LPPau0vKytDaGio1r5OnTppbRcVFWH+/Pn4/vvvkZmZiYqKCty6dQtpaWk6xXD+/HlMnDhRa1+3bt2wevVqrX2aflMA4OjoCBcXF+Tk5AAAoqOjMXToUJw6dQrh4eEYNGgQunbtqlMcRPRgTICI6IGioqIwadIkrF27FrGxsWjVqhV69OghH2/dujV++uknlJWVVUt0MjIyUFBQgEcffVRrf4sWLfQ6hLyoqAgA8P333+ORRx7ROmZra6u17ejoqLU9ffp0xMfHY8WKFQgICIC9vT2ee+45lJWV6S2+qu5O9iRJglqtBgD06dMHV65cwQ8//ID4+Hj07NkTMTExWLFihUFiIbJUHAVGRA/0/PPPw8rKCtu2bcPWrVvx0ksvQZIk+fjIkSNRVFSETz75pNq1K1asgJ2dHYYPH663eDRJlkqlkvcFBQXB1tYWaWlpCAgI0Po0bdr0vvf7+eefMXbsWAwePBjBwcHw8fHB5cuXqz2z6vNq0qZNG/z888/V7h0UFKRD6QBPT09ERkbis88+w6pVq7BhwwadrieiB2MNEBE9kJOTE4YPH45Zs2ahoKAAY8eO1ToeFhaGKVOmYMaMGSgrK8OgQYNQXl6Ozz77DB9++CE2b94MDw8PrWtycnJQUlKitc/Dw6NWTWH+/v6QJAlxcXHo27cv7O3t4ezsjOnTp2Pq1KlQq9Xo3r078vPz8fPPP8PFxQWRkZH3vF/r1q2xY8cO9O/fH5IkYc6cOXKNjEbz5s1x+PBhjBgxAra2tmjcuHG1+8yYMQPPP/88QkND0atXL3z33XfYsWMH9u7d+8AyacydOxcdO3ZE27ZtUVpairi4OLRp06bW1xNRLZm6ExIRNQxHjx4VALQ69N5t48aNomPHjsLOzk4AEEqlUhw6dEjrHE1H5Jo+CQkJNd737k7QQgixcOFC4ePjIyRJkofBq9VqsWrVKvHYY48JhUIhPD09RUREhBzDvUagpaamiqefflrY29uLpk2bijVr1ogePXqIKVOmyOckJCSIkJAQYWtr+9DD4Hfu3Km1z9XVVcTGxgohhFi0aJFo06aNsLe3F+7u7mLgwIHi0qVL93jjRFRXkhAP6LlIRFQHly9fRo8ePRAWFobPP/8c1tbWpg6JiEjGPkBEZBDNmzfHwYMHERgYKC9hQURUX7AGiIiIiCwOa4CIiIjI4jABIiIiIovDBIiIiIgsDhMgIiIisjhMgIiIiMjiMAEiIiIii8MEiIiIiCwOEyAiIiKyOEyAiIiIyOL8PzAhpzTf9hrXAAAAAElFTkSuQmCC\n",
```

Comment:
Seeing this image it looks like the VQE does an awful lot of unnecessary iterations. Can't we shorten it to ~50-75 cycles to keep the tutorial runtime down?

Reply:
yes we can do it, but sometimes (due to cvar and qasm simulator you need more iterations so I would prefer to keep it to 200).

Reply:
The tutorial should be run with the `algorithm_seed` specified. Otherwise it will keep changing it's values everytime its run by the CI. That is undesirable.

Reply:
I think the changing behavior is because of the fact that the Paulis are grouped and the grouping algorithm was found to be non-deterministic in terms of the order of the groups - see Qiskit/retworkx#347. The issue has been closed but looking at dates it does not look like its in a released build yet. So I think in the interim we will have to accept that result varies (we noticed originally in some unit test we have which would occasionally fail and it turned out to be because the order of the groups varies (hence the ccts are executed in different order and we do not get quite the same outcome)